advanced/wallets/react-wallet-v2/src/lib/KernelSmartAccountLib.ts

```diff
@@ -0,0 +1,218 @@
+import { Chain } from '@/utils/SmartAccountUtils'
+import { Address, createPublicClient, Hex, http, parseEther, PrivateKeyAccount, PublicClient, zeroAddress } from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import { EIP155Wallet } from './EIP155Lib'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { KernelValidator } from '@zerodev/ecdsa-validator'
+import {
+    addressToEmptyAccount,
+    createKernelAccount,
+    createKernelAccountClient,
+    createZeroDevPaymasterClient,
+    KernelAccountClient
+} from '@zerodev/sdk'
+import { sepolia } from 'viem/chains'
+import { serializeSessionKeyAccount, signerToSessionKeyValidator } from '@zerodev/session-key'
+import { createWeightedECDSAValidator, getUpdateConfigCall } from '@zerodev/weighted-ecdsa-validator'
+import { BundlerActions, bundlerActions, BundlerClient } from 'permissionless'
+
+type SmartAccountLibOptions = {
+    privateKey: string
+    chain: Chain
+    sponsored?: boolean
+  };
+  
+
+export class KernelSmartAccountLib implements EIP155Wallet {
+    public chain: Chain
+    public isDeployed: boolean = false;
+    public address?: `0x${string}`;
+    public sponsored: boolean = true;
+    private signer: PrivateKeyAccount;
+    private client: KernelAccountClient | undefined
+    private publicClient: PublicClient & BundlerClient & BundlerActions| undefined
+    private validator: KernelValidator | undefined
+
+
+    #signerPrivateKey: string;
+
+    public constructor({ privateKey, chain, sponsored = false }: SmartAccountLibOptions) {        
+        this.chain = chain
+        this.sponsored = sponsored
+        this.#signerPrivateKey = privateKey
+        this.signer = privateKeyToAccount(privateKey as Hex)
```

Comment:
for future: if Kernel allows signers, a couple of static methods to construct from signers or PK would be nice